WPFHospitalEditor/SpecialistAppointmentSearchDialog.xaml.cs

```diff
@@ -0,0 +1,135 @@
+﻿using HealthcareBase.Dto;
+using HealthcareBase.Model.Schedule.SchedulingPreferences;
+using HealthcareBase.Model.Users.Employee.Doctors;
+using HealthcareBase.Model.Users.Employee.Doctors.DTOs;
+using HealthcareBase.Model.Utilities;
+using HospitalWebApp.Dtos;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Windows;
+using WPFHospitalEditor.Controller;
+using WPFHospitalEditor.Controller.Interface;
+
+namespace WPFHospitalEditor
+{
+    /// <summary>
+    /// Interaction logic for SpecialistAppointmentSearchDialog.xaml
+    /// </summary>
+    public partial class SpecialistAppointmentSearchDialog : Window
+    {
+        IDoctorServerController doctorServerController = new DoctorServerController();
+        IEquipmentTypeServerController equipmentTypeServerController = new EquipmentTypeServerController();
+        IEquipmentServerController equipmentServerController = new EquipmentServerController();
+        ISchedulingServerController schedulingServerController = new SchedulingServerController();
+        HospitalMap hospitalMap;
+
+        public SpecialistAppointmentSearchDialog(HospitalMap hospitalMap)
+        {
+            InitializeComponent();
+            this.hospitalMap = hospitalMap;
+            loadSpecialists();
+            loadEquipment();
+        }
+
+        private void loadSpecialists()
+        {
+            List<DoctorDto> specialists = doctorServerController.GetAllSpecialists().ToList();
+            foreach(DoctorDto d in specialists)
+            {
+                specialistComboBox.Items.Add(d.DoctorId.ToString() + " " + d.Name + " " + d.Surname + " [" + d.DepartmentName + "]");
+            }
+            
+        }
+
+        private void loadEquipment()
+        {
+            List<EquipmentTypeDto> equipmentTypes = equipmentTypeServerController.GetAllEquipmentTypes().ToList();
+            
+            foreach(EquipmentTypeDto e in equipmentTypes)
+            {
+                equipmentComboBox.Items.Add(e.Name);
+            }
+        }
+
+        private void Close_Click(object sender, RoutedEventArgs e)
+        {
+            this.Close();
+        }
+
+        private void Search_Click(object sender, RoutedEventArgs e)
+        {
+            HospitalMap.clearAllResults();
+            if (InvalidInputForAppointment())
+            {
+                MessageBox.Show("Invalid input.");
+            }
+            else
+            {
+                String specialist = specialistComboBox.SelectedItem.ToString();
```

Comment:
The problem of using a combo box to display a list of data, that can be quite long, reappears. Instead of refactoring later, why not change it immediately and make it a more practical way?